pinot-core/src/main/java/com/linkedin/pinot/server/realtime/ServerSegmentCompletionProtocolHandler.java

```diff
@@ -96,130 +59,100 @@ public InputStream createInputStream() throws IOException {
     params.withInstanceId(_instanceId).withOffset(offset).withSegmentName(segmentName);
     SegmentCompletionProtocol.SegmentCommitUploadRequest request = new SegmentCompletionProtocol.SegmentCommitUploadRequest(params);
 
-    return postSegmentToVip(request, createFileParts(segmentName, segmentTarFile), controllerVipUrl);
+    String hostPort;
+    String protocol;
+    try {
+      URI uri = URI.create(controllerVipUrl);
+      protocol = uri.getScheme();
+      hostPort = uri.getAuthority();
+    } catch (Exception e) {
+      throw new RuntimeException("Could not make URI", e);
+    }
+    String url = request.getUrl(hostPort, protocol);
```

Comment:
Why not directly use the uri?

Reply:
Currently the request object returns a url string with all the parameters.